src/riak_kv_pb_bucket.erl

```diff
@@ -66,90 +66,87 @@ init() ->
-                           <<"default">>;
-                       T ->
-                           T
-                   end,
-            {ok, Msg, {"riak_kv.list_buckets", Type}};
-        #rpblistkeysreq{} ->
-            Type = case Msg#rpblistkeysreq.type of
-                       undefined ->
-                           <<"default">>;
-                       T ->
-                           T
-                   end,
-            {ok, Msg, {"riak_kv.list_keys", {Type,
-                                                Msg#rpblistkeysreq.bucket}}}
-    end.
+    handle_decoded(Msg).
+
+handle_decoded(rpblistbucketsreq = Msg) ->
+    %% backwards compat
+    {ok, Msg, {"riak_kv.list_buckets", <<"default">>}};
+handle_decoded(#rpblistbucketsreq{} = Msg) ->
+    Type = convert_type(Msg#rpblistbucketsreq.type),
+    {ok, Msg, {"riak_kv.list_buckets", Type}};
+handle_decoded(#rpblistkeysreq{} = Msg) ->
+    Type = convert_type(Msg#rpblistkeysreq.type),
+    {ok, Msg, {"riak_kv.list_keys", {Type,
+                                     Msg#rpblistkeysreq.bucket}}}.
 
 %% @doc encode/1 callback. Encodes an outgoing response message.
 encode(Message) ->
     {ok, riak_pb_codec:encode(Message)}.
 
-%% @doc process/2 callback. Handles an incoming request message.
-process(#rpblistbucketsreq{type = Type, timeout = T, stream = S} = Req, State) ->
-    Class = case S of
-        true ->
-            {riak_kv, stream_list_buckets};
-        _ ->
-            {riak_kv, list_buckets}
-    end,
-    Accept = riak_core_util:job_class_enabled(Class),
-    _ = riak_core_util:report_job_request_disposition(
-            Accept, Class, ?MODULE, process, ?LINE, protobuf),
-    case Accept of
-        true ->
-            case check_bucket_type(Type) of
-                {ok, GoodType} ->
-                    do_list_buckets(GoodType, T, S, Req, State);
-                error ->
-                    {error,
-                        {format, "No bucket-type named '~s'", [Type]}, State}
-            end;
-        _ ->
-            {error,
-                riak_core_util:job_class_disabled_message(binary, Class),
-                State}
-    end;
 
 %% this should remain for backwards compatibility
 process(rpblistbucketsreq, State) ->
     process(#rpblistbucketsreq{}, State);
 
-%% Start streaming in list keys
-process(#rpblistkeysreq{type = Type, bucket = B, timeout = T} = Req,
-        #state{client = Client} = State) ->
-    % at present list-keys always streams
-    Class = {riak_kv, stream_list_keys},
+%% @doc process/2 callback. Handles an incoming request message.
+process(Req, State) ->
+    {Class, Listing} = determine_class_and_listing(Req),
+    Accept = determine_accept_and_report_job_disposition(Class),
+    case {Accept, Listing} of
+        {true, buckets} ->
+            maybe_do_list_buckets(Req, State);
+        {true, keys} ->
+            %% at present list-keys always streams
+            %% Start streaming in list keys
```

Comment:
I would remove this "Start streaming in list keys" comment. Doesn't seem to really add anything, given the next function is called "maybe_stream_list_keys".